src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -392,6 +391,10 @@ private static unsafe int AlpnServerSelectCallback(IntPtr ssl, out byte* outp, o
                 return Ssl.SSL_TLSEXT_ERR_NOACK;
```

Comment:
Need to null out the target here as well because it swallows the exception and exists here.

Reply:
If you change it to free, then this should read "you need to free out the handle here"

Reply:
If the code throws here, it is the case of server not supported alpn, rather than server supporting alpn and was not able to find a match with client protocols.

Reply:
If the code throws in there, its actually that there was a buffer over/underrun. I put the try/catch in to catch any Slice operations that will fail range checks. I am not sure how you want to report that back to the user. If its just an okay no match then that is fine. It shouldn't even get into the callback if alpn is unsupported?

Reply:
Right, the callback is not called if alpn is not supported, but when the slice operations fail on server side, it is the case of some error on the server, so from client's perspective server doesn't support alpn, rather than server supported alpn and the matching failed.

Reply:
But it slices the client buffer, so its most likely the client passed a bad buffer, the servers protocols are a typesafe list, eg client sends a buffer with a length prefix for a protocol that is too long or short. 

But either way, whatever you think the correct behaviour is that is fine.

Reply:
Hmm, i missed that it's the client protocol list that throws, in that case, i think it should be the non matched behavior rather than not supported. The not supported should be the behavior when server protocol list is invalid, which would be the case when the gchandle target doesn't hold the right value.